### PR TITLE
Enable local doc building without git installation

### LINKED ISSUE #13013: v3.0.2 local html docs "git describe" error
### Bug report

**Bug summary**

*make html* fails when building html docs from downloaded source (https://github.com/matplotlib/matplotlib/archive/v3.0.2.zip), probably, because *.git* directory is not included there. The intention is to build docs for offline use.

**Code for reproduction**

```bash
make html
```

**Actual outcome**

```
Running Sphinx v1.7.6
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

Configuration error:
There is a programable error in your configuration file:

Traceback (most recent call last):
  File "/home/<username>/.local/lib/python3.6/site-packages/sphinx/config.py", line 161, in __init__
    execfile_(filename, config)
  File "/home/<username>/.local/lib/python3.6/site-packages/sphinx/util/pycompat.py", line 150, in execfile_
    exec_(code, _globals)
  File "conf.py", line 145, in <module>
    SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
  File "/usr/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'describe', '--dirty']' returned non-zero exit status 128.

Makefile:20: recipe for target 'html' failed
make: *** [html] Error 2
```

**Expected outcome**

A set of html docs.

**Matplotlib version**

  * Operating system: Ubuntu 18.04.1 LTS
  * Matplotlib version: 3.0.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): agg
  * Python version: Python 3.6.6
  * Jupyter version (if applicable):
  * Other libraries: 

**Workaround**

Modify *conf.py*:

```python
#SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
SHA = "8858a0d1bdd149a0897789e8503ac586be14676d"
```

### PR DESCRIPTION
**Pull request by dstansby**
This just uses the version that `versioneer` gives Matplotlib. This should be a 1:1 mapping between commits and version numbers (my local build gives ` Doc version 3.0.2.post1037.dev0+g0fec7f1c3.`).

Fixes #13013

### CODE REVIEW

**Comment by dstansby** [comment]
Ah fair enough, maybe adding a `try/except` block to the git command would work instead?

**Comment by tacaswell** [comment]
I think the best fix would be to replicate part of versioneer and use the git export templating, a quick fix would be to put the subprocess call in a `try..except` block and fallback to using `matplotlib.__version__`.

**Comment by tacaswell** [comment]
Our last two comments passed in the ether, looks like `try...except` is the winner.

File: doc/conf.py [hunk 1]
```diff
@@ -141,8 +142,13 @@ def _check_deps():
...
 
 # General substitutions.
-from subprocess import check_output
-SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
+try:
+    SHA = subprocess.check_output(
+        ['git', 'describe', '--dirty']).decode('utf-8').strip()
+# Catch the case where git is not installed locally, and use the versioneer
+# version number instead
+except subprocess.CalledProcessError:
```
Comment (anntzer) [inline doc/conf.py]: actually that raises a filenotfounderror, not a calledprocesserror, when git is not in the $PATH

**Pull request merged**